app/controllers/changes_controller.rb

```diff
@@ -23,6 +23,34 @@ def resend
     render json: @change.send_notifications
   end
 
+  def html
+    @change_id=params[:change_id]
+    @snapshot = PageSnapshot.find(@change_id)
+    @page = Page.find(@snapshot.page_id)
```

Comment:
Similarly this can be simplified using ActiveRecord relations to `@snapshot.page`. Also just as a general note if a piece of data is a simple relation, you don't really need to assign it to a controller variable, you can just reference the relation in the view. This can get you into trouble with complex or expensive relational queries, but in general Rails does a good job of smart caching and you shouldn't worry too much about it.

Reply:
I'm not aware of this so will have to read up on what ActiveRecord relations are, but I think I generally understand what you're saying here.

Reply:
Don't worry about going too deep, you can just access related data in other tables through methods on the model.